protocol/protocol.go

```diff
@@ -126,6 +127,34 @@ func New(config ProtocolConfig) *Protocol {
 	return p
 }
 
+// CurrentState returns the current protocol state
+func (p *Protocol) CurrentState() State {
+	stateChan := make(chan State, 1)
+	p.stateTransitionChan <- protocolStateTransition{
+		stateRespChan: stateChan,
+	}
+	return <-stateChan
+}
```

Comment:
There's probably a cleaner way to do this. Instead, let's make `currentState` defined at the top of `stateLoop()` a field in the `Protocol` type, and then we can access it more directly.